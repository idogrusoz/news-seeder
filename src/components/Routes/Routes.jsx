import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';
import Headlines from '../Headlines/Headlines';

const Routes = ({ history }) => {
    return (
        <Router history={history}>
            <Switch>
                <Route exact path="/" component={Headlines} />
            </Switch>
        </Router>
    );
};

export default Routes;
