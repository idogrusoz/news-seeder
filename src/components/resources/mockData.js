import { act } from '@testing-library/react';

export let mockNews = [
    {
        source: {
            id: null,
            name: 'YouTube',
        },
        author: null,
        title: 'mock-title',
        description: 'mock-description',
        url: 'https://www.mock-url.be',
        urlToImage: 'https://mock-url.be/maxresdefault.jpg',
        publishedAt: '2020-12-21T08:00:30Z',
        content: null,
    },
    {
        source: {
            id: null,
            name: 'YouTube',
        },
        author: null,
        title: 'mock-title',
        description: 'mock-description',
        url: 'https://www.mock-url2.be',
        urlToImage: null,
        publishedAt: '2020-12-21T08:00:30Z',
        content: 'some text <script>danger</script> [+42]',
    },
];

export const waitForComponentToPaint = async (wrapper) => {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, 0));
        wrapper.update();
    });
};
