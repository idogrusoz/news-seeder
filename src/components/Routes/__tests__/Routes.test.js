import { mount } from 'enzyme';
import { createMemoryHistory } from 'history';
import { mockNews } from '../../resources/mockData';
import Routes from '../Routes';

describe('Routes test', () => {
    it('renders Headlines on root', () => {
        const history = createMemoryHistory({ initialEntries: ['/'] });
        const wrapper = mount(<Routes history={history} />);
        expect(wrapper.find('Headlines').length).toBeGreaterThan(0);
    });
    it('reners Articles on /articles root', () => {
        const history = createMemoryHistory({ initialEntries: ['/article'] });
        const wrapper = mount(<Routes history={history} selected={mockNews[0]} />);
        expect(wrapper.find('Article').length).toBeGreaterThan(0);
    });
});
