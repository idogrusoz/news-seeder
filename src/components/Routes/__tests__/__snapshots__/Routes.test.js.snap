// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "height:100vh",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "style": "width:100%;height:100%;display:flex;justify-content:center;align-items:center",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "css-3fvcat",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "css-1s6oywz",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "css-6eg5q1",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "css-6eg5q1",
                },
                "children": Array [],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "css-1s6oywz",
                  },
                  "children": Array [],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
